src/api/registries/mutate/createRegistry.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  RegistryCreatedSchema,
+  RegistryCreatedSchemaType,
+  RegistryFormSchemaType,
+} from "../schema";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { apiFactory } from "~/api/utils";
+import { registriesKeys } from "..";
+import { useApiKey } from "~/util/store/apiKey";
+import { useNamespace } from "~/util/store/namespace";
+import { useToast } from "~/design/Toast";
+
+const createRegistry = apiFactory({
+  url: ({ namespace }: { namespace: string }) =>
+    `/api/functions/registries/namespaces/${namespace}`,
+  method: "POST",
+  schema: RegistryCreatedSchema,
+});
+
+export const useCreateRegistry = ({
+  onSuccess,
+}: {
+  onSuccess?: (registry: RegistryCreatedSchemaType) => void;
+} = {}) => {
+  const apiKey = useApiKey();
+  const namespace = useNamespace();
+  const { toast } = useToast();
+  const queryClient = useQueryClient();
+
+  if (!namespace) {
+    throw new Error("namespace is undefined");
+  }
+
+  const mutationFn = ({ url, user, password }: RegistryFormSchemaType) =>
+    createRegistry({
+      apiKey: apiKey ?? undefined,
+      payload: { data: `${user}:${password}`, reg: url },
+      urlParams: {
+        namespace: namespace,
+      },
+      headers: undefined,
+    });
+
+  return useMutation({
+    mutationFn,
+    onSuccess: (registry, variables) => {
+      queryClient.invalidateQueries(
```

Comment:
I postponed implementing an optimistic cache update for later because I wanted to focus on getting this to work first.

Now that it works, I somehow prefer keeping it like this. The code is cleaner, less complex, and refetching the data shouldn't have a large performance impact?

Reply:
I agree that it is much cleaner and it's good (at least for now) to keep it simple and just refetch. 

Performance shouldn't be the reason for optimistic updates, I think it's more about skipping a loading state and make the UI a bit more instant. We might even decide to just opt in to optimistic updates when it's about a feature where the user interacts a lot.